### PR TITLE
Check IES GFX support

### PR DESCRIPTION
**Pull request by pH5**
dgcgears [doesn't work on radv](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/31383#note_2586274).

Check for support to avoid running into a assertions:
```
dgcgears: ../src/amd/vulkan/radv_dgc.c:3159: radv_CreateIndirectExecutionSetEXT: Assertion `pipeline->type == RADV_PIPELINE_COMPUTE' failed.
```

### CODE REVIEW

File: src/dgcgears.c [hunk 1]
```diff
@@ -1760,6 +1760,24 @@ check_sample_count_support(VkSampleCountFlagBits sample_count)
...
+      VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DEVICE_GENERATED_COMMANDS_PROPERTIES_EXT,
+   };
+   VkPhysicalDeviceProperties2 properties = {
+      VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
+      .pNext = &dgcproperties,
+   };
+   vkGetPhysicalDeviceProperties2(physical_device, &properties);
+
+   if (use_shader_object)
+      return dgcproperties.supportedIndirectCommandsShaderStagesShaderBinding & VK_SHADER_STAGE_FRAGMENT_BIT;
```
Comment (zmike) [inline src/dgcgears.c:1776]: These should both be checking VS+FS.

Comment (pH5) [reply src/dgcgears.c]: Whoops, fixed.

File: src/dgcgears.c [hunk 2]
```diff
@@ -1898,6 +1917,9 @@ main(int argc, char *argv[])
    if (!check_sample_count_support(sample_count))
       error("Sample count not supported");
 
+   if (!check_indirect_commands_graphics_support())
+      error("Indirect execution does not support graphics %s", use_shader_object ? "shader" : "pipeline");
```
Comment (zmike) [inline src/dgcgears.c:1921]: Also this should say `does not support graphics X switching`

Comment (pH5) [reply src/dgcgears.c]: Done.

Comment (zmike) [reply src/dgcgears.c]: Great, thanks

**Pull request merged**